net/nts/nts.go

```diff
@@ -202,33 +187,36 @@ func DecodePacket(pkt *NTSPacket, b []byte, key []byte) (err error) {
 	return nil
 }
 
-func ExtractCookie(b []byte) ([]byte, error) {
-	msgbuf := bytes.NewReader(b[48:])
-	for msgbuf.Len() >= 28 {
+func (pkt *NTSPacket) Authenticate(b *[]byte, key []byte) error {
```

Comment:
Is `Authenticate` a good name for this method? It also decrypts, right? What about the little less specific name `DecodeCookies`?

Reply:
Tough one. Maybe we could also introduce a new function nts.ProcessRequest, make nts.NTSPacket.Authenticate private and call it from nts.ProcessRequest and from nts.ProcessResponse. This way, the entire interface would also get a more symmetrical.

Reply:
By the way: the type should be named `nts.Packet` and not `nts.NTSPacket`.

Reply:
For the response side, it wouldn't make much of a difference because the two calls happen in sequence anyway, right? And for the request side we have only one call so far. If this is correct, the new idea would be fine with me.

Reply:
Done